src/server/middleware/StaticAssetHandler.ts

```diff
@@ -29,22 +29,24 @@ export class StaticAssetHandler extends HttpHandler {
    *  where URL paths ending in a slash are interpreted as entire folders.
    * @param options - Cache expiration time in seconds.
    */
-  public constructor(assets: Record<string, string>, options: { expires?: number } = {}) {
+  public constructor(assets: Record<string, string>, baseUrl: string, options: { expires?: number } = {}) {
     super();
     this.mappings = {};
+    const rootPath = new URL(baseUrl).pathname;
+
     for (const [ url, path ] of Object.entries(assets)) {
-      this.mappings[url] = resolveAssetPath(path);
+      this.mappings[url.replace(/^\//u, rootPath)] = resolveAssetPath(path);
     }
-    this.pathMatcher = this.createPathMatcher(assets);
+    this.pathMatcher = this.createPathMatcher();
```

Comment:
Seeing as all URLs will start with the same root path, I would not use the long versions of the URLs but instead change how the regex is made.

For all the URLs in `assets` I would strip the leading slash (could add utility function `trimLeadingSlashes`, similar to how we have `ensureLeadingSlash` and `trimTrailingSlashes`). The path matcher regex could then be changed to `^${rootPath}(?:(${files.join('|')})|(${folders.join('|')})([^?]+))(?:\\?.*)?$`, making sure the `rootPath` has a trailing slash.

So `createPathMatcher` would take an additional `rootPath` parameter besides `assets`.

Reply:
Is there a reason why in the initial code the assets record gets passed to `createPathMatcher` instead of using `this.mappings` in that function? Since the URLs are just copied from one Record to the other and the function only uses those keys. (This is how I have implemented the changes now btw)

I'll implement above changes but is it OK if I use `trimLeadingSlashes` on the url keys for mappings then use that to create the PathMatcher?

Reply:
> Is there a reason why in the initial code the assets record gets passed to `createPathMatcher` instead of using `this.mappings` in that function?

No special reason I think.



> I'll implement above changes but is it OK if I use `trimLeadingSlashes` on the url keys for mappings then use that to create the PathMatcher?

Yes that's probably better actually, since otherwise the `getFilePath` function wouldn't work any more.